tests/utils.py

```diff
@@ -34,12 +41,15 @@ def __init__(self, handler, *args, **kwargs):
 
 def create_server(thread_class):
     def server(handler):
-        server_thread = thread_class(handler, daemon=True)
-        server_thread.start()
-        try:
-            yield server_thread
-        finally:
-            server_thread.terminate()
+        lock = filelock.FileLock(LOCK_PATH)
+        with lock:
+            server_thread = thread_class(handler, daemon=True)
+            server_thread.start()
+            try:
+                yield server_thread
+            finally:
+                server_thread.terminate()
+            os.remove(LOCK_PATH)
```

Comment:
I don't think so, there will be an exception raised and the locked scoped will be left.

Reply:
But the removal is problematic for Windows:
```
>           os.remove(LOCK_PATH)
E           PermissionError: [WinError 32] The process cannot access the file because it is being used by another process: 'D:\\a\\sphinx\\sphinx\\tests\\test-server.lock'
```

so we should likely leave the file there.

Reply:
Yes, that's what I thought, but I am wondering how the `filelock` API *acquires* its lock in the first place. Anyway, assuming that it works, are you sure that the port will be freed at the same time that the lock is released ?

Reply:
Yes, I made an intensive parallel testing `pytest tests/test_build_linkcheck.py -n 32` and it works fine now.

Reply:
> Yes, I made an intensive parallel testing `pytest tests/test_build_linkcheck.py -n 32` and it works fine now.

What I actually fear is that the behaviour is different on Linux and Windows builds. Therefore, I am wondering whether we could add a specific "optional" CI checks that would do that check for us on those builds.

Reply:
What type of check do you speak about?

Reply:
This one: `pytest tests/test_build_linkcheck.py -n 32`

Reply:
Sure, let me write it on my TODO-list. Yes, we can do that once we'll enable parallel tests generally.